client/src/components/common/task-solutions/index.tsx

```diff
@@ -4,22 +4,24 @@ import moment from 'moment';
 import SyntaxHighlighter from 'react-syntax-highlighter/dist/esm/default-highlight';
 import { Rank } from 'components/basic';
 import BetaRank from 'components/basic/rank/beta';
-import { TTaskSolutions } from 'containers/profile/tabs/solutions/mocks';
 import { useThemeSelector } from 'hooks/useAppSelector';
 import { ThemeType } from 'containers/theme-switcher/logic/models';
 import { ROUTES } from 'constants/routes';
 import { TaskStatus } from 'typings/common/task';
 import { SolutionStatus } from 'typings/common/solution';
 import styles from './task-solutions.module.scss';
 import { darkTheme, lightTheme } from './config';
+import { WebApi } from 'typings/webapi';
 
 interface ITaskSolutionsProps {
-	taskSolutions: TTaskSolutions;
+	task: WebApi.Entities.ITask;
 }
 
 function readableLanguageFormat(language: string) {
 	switch (language) {
-		case 'javascript': {
+		case 'javascript':
+		case undefined: {
+			// TODO: remove undefined
```

Comment:
do we need this undefined case? why not remove it?

Reply:
`Solution` entity doesn't have `language` property for now, so it's just `'JavaScript'`